### ISSUE TITLE
Internationalize datatable text

### ISSUE THREAD

**Issue by Bubballoo3** [OSC/ondemand#5074]
Noticed during #5046, we pass some hard coded strings like "filter" to all of the tables. This should be internationalized as all the table text will be the same on all tables

**Comment by Bubballoo3** [comment]
I initially did this for the files app 'Filter' text, but this should have a wider scope than that. The general approach I found to work was the inverse of this diff

```
diff --git a/apps/dashboard/app/javascript/files/data_table.js b/apps/dashboard/app/javascript/files/data_table.js
index 85a25f49..3b9d14fe 100644
--- a/apps/dashboard/app/javascript/files/data_table.js
+++ b/apps/dashboard/app/javascript/files/data_table.js
@@ -189,7 +189,7 @@ class DataTable {
         }).DataTable({
             autoWidth: false,
             language: {
-                search: `${history.state.filterText}:`,
+                search: 'Filter:',
                 infoFiltered: ""
             },
             order: [[1, "asc"], [2, "asc"]],
diff --git a/apps/dashboard/app/javascript/files/page_load.js b/apps/dashboard/app/javascript/files/page_load.js
index 98c09117..7351abdf 100644
--- a/apps/dashboard/app/javascript/files/page_load.js
+++ b/apps/dashboard/app/javascript/files/page_load.js
@@ -14,7 +14,6 @@ export function setPageLoadState() {
     currentDirectoryUpdatedAt: data['currentDirectoryUpdatedAt'],
     currentFilesPath: data['currentFilesPath'],
     currentFilesUploadPath: data['currentFilesUploadPath'],
-    currentFilesystem: data['currentFilesystem'],
-    filterText: data['filterText']
+    currentFilesystem: data['currentFilesystem']
   }, null);
 }
diff --git a/apps/dashboard/app/views/files/_page_load_config.html.erb b/apps/dashboard/app/views/files/_page_load_config.html.erb
index 685dfb40..d2c96cfc 100644
--- a/apps/dashboard/app/views/files/_page_load_config.html.erb
+++ b/apps/dashboard/app/views/files/_page_load_config.html.erb
@@ -7,7 +7,6 @@
   data-current-filesystem="<%= @filesystem %>"
   data-download-error-title="<%= t('dashboard.files_directory_download_error_modal_title') %>"
   data-current-files-upload-path="<%= url_for(fs: @filesystem, action: 'upload') if Configuration.upload_enabled? %>"
-  data-filter-text="<%= I18n.t('dashboard.filter') %>"
   >
 
 </div>
diff --git a/apps/dashboard/config/locales/en.yml b/apps/dashboard/config/locales/en.yml
index 266fdab3..ffeabc88 100644
--- a/apps/dashboard/config/locales/en.yml
+++ b/apps/dashboard/config/locales/en.yml
@@ -136,7 +136,6 @@ en:
     directory: Directory
     edit: Edit
     file: File
-    filter: Filter
     file_quotas: File Quotas
     files_directory_download_error_modal_title: Directory too large to download
     files_directory_download_unauthorized: You can only download a file or directory that you have read and execute access to
```

**Comment by moffatsadeghi** [comment]
GitHub Review Meeting Summary: 
Discussion about internationalizing text in data tables, dependent on JavaScript internationalization infrastructure.
Details
    Braeden: Almost completed during data tables upgrade but deferred; found workaround with config strings but prefers proper JavaScript internationalization
    johrstrom: Created parent ticket for JavaScript internationalization; established parent-child relationship
    johrstrom: Believes screen readers may already translate known words based on language settings
    Braeden: Marked as accessibility but acknowledged it's not WCAG-related
Conclusion
    Move to backlog
    Not essential for 4.2 release
